test/execution/atomics_test.cpp

```diff
@@ -0,0 +1,513 @@
+#include <random>
+#include <string>
+
+#include "common/macros.h"
+#include "common/worker_pool.h"
+#include "execution/ast/ast_node_factory.h"
+#include "execution/ast/context.h"
+#include "execution/compiler/codegen.h"
+#include "execution/compiler/compiler.h"
+#include "execution/compiler/function_builder.h"
+#include "execution/sema/error_reporter.h"
+#include "execution/tpl_test.h"
+#include "execution/util/region.h"
+#include "execution/util/region_containers.h"
+#include "execution/vm/module.h"
+#include "execution/vm/vm_defs.h"
+#include "test_util/multithread_test_util.h"
+
+namespace noisepage::execution::test {
+
+class AtomicsTest : public TplTest {
+ public:
+  AtomicsTest() : region_("atomics_test"), pos_() {}
+
+  util::Region *Region() { return &region_; }
+
+  const SourcePosition &EmptyPos() const { return pos_; }
+
+  util::Region region_;
+  SourcePosition pos_;
+};
+
+// NOLINTNEXTLINE
+TEST_F(AtomicsTest, AtomicAndOr1) {
+  // Setup the compilation environment
+  sema::ErrorReporter error_reporter(&region_);
+  ast::AstNodeFactory factory(&region_);
+  ast::Context context(&region_, &error_reporter);
+
+  auto src = std::string(
+      "fun atomic_and(dest: *uint8, mask: uint8) -> uint8 {\n"
+      "  var x = @atomicAnd(dest, mask)\n"
+      "  return x\n"
+      "}\n"
+      "fun atomic_or(dest: *uint8, mask: uint8) -> uint8 {\n"
+      "  var x = @atomicOr(dest, mask)\n"
+      "  return x\n"
+      "}");
+
+  // Compile it...
+  auto input = compiler::Compiler::Input("Atomic Definitions", &context, &src);
+  auto module = compiler::Compiler::RunCompilationSimple(input);
+  EXPECT_STREQ(error_reporter.SerializeErrors().c_str(), "");
+  ASSERT_FALSE(module == nullptr);
+
+  // The function should exist
+  std::function<uint8_t(uint8_t *, uint8_t)> atomic_and;
+  EXPECT_TRUE(module->GetFunction("atomic_and", vm::ExecutionMode::Interpret, &atomic_and));
```

Comment:
Have you tested these in compiled mode, too?

Reply:
I have not, but none of the execution unittests use compiled mode at the moment.

Reply:
We have an issue for that [here](https://github.com/cmu-db/noisepage/issues/1165). In the mean time, check here to be sure.

Reply:
Acknowledge.  If you're fine with the general style of the test code now, I'll add a parameter to the templated functions to indicate whether or not to compile the code.  And test everything in both interpreted and compiled mode.

Reply:
Okay with me.

Reply:
I am able to run the tests in compiled mode as long as I copy the `bytecode_handlers_ir.bc` file into my working directory before execution.  This seems like there is a larger issue with how that bytecode is stored and loaded into the library that is mostly outside the scope of this PR.

What are your thoughts on having the templated tests instrumented for both execution modes but commenting out the compiled tests until we settle on the right way to handle that file dependency?